shiny/experimental/ui/_valuebox.py

```diff
@@ -0,0 +1,186 @@
+from __future__ import annotations
+
+import numbers
+from typing import Callable, Optional
+
+from htmltools import Tag, TagAttrs, TagChild, css, div
+
+from shiny._typing_extensions import TypeGuard
+
+from ._card import card, card_body
+from ._card_item import CardItem
+from ._css import CssUnit, validate_css_unit
+from ._fill import bind_fill_role
+from ._layout import layout_column_wrap
+
+
+def is_01_scalar(x: object) -> TypeGuard[float]:
+    return isinstance(x, float) and x >= 0.0 and x <= 1.0
+
+
+# It seems to be to use % over fr here since there is no gap on the grid
+def validate_width_unit(x: str | float) -> str:
+    if is_01_scalar(x):
+        scaled_x = 100 * x
+        scaled_not_x = 100 * (1 - x)
+        return f"{scaled_x}% {scaled_not_x}%"
+
+    # TODO: validateCssUnit() should maybe support fr units?
+    # return(paste(x, collapse = " "))
+    return str(x)
+
+
+def validate_height_unit(x: str | float) -> str:
+    if is_01_scalar(x):
+        return f"{100 * x}%"
+
+    height_unit = validate_css_unit(x)
+    assert height_unit is not None
+    return height_unit
+
+
+def value_box(
+    title: TagChild | str | numbers.Number,
+    value: TagChild | str | numbers.Number,
+    *args: TagChild | TagAttrs,
+    showcase: Optional[TagChild] = None,
+    showcase_layout: Callable[[TagChild, Tag], CardItem] | None = None,
+    full_screen: bool = False,
+    theme_color: Optional[str] = "primary",
+    height: Optional[CssUnit] = None,
+    max_height: Optional[CssUnit] = None,
+    fill: bool = True,
+    class_: Optional[str] = None,
+    **kwargs: TagChild | TagAttrs,
+) -> Tag:
+    if showcase_layout is None:
+        showcase_layout = showcase_left_center()
+    if isinstance(title, str) or isinstance(title, numbers.Number):
+        title = div(str(title), class_="h6 mb-1")
+    if isinstance(value, str) or isinstance(value, numbers.Number):
+        value = div(str(value), class_="h2 mb-2")
+
+    attribs = kwargs
+    children = args
+
+    contents = div(
+        title,
+        value,
+        *children,
+        class_="value-box-area",
+    )
+    contents = bind_fill_role(contents, container=True, item=True)
+
+    if showcase is not None:
+        contents = showcase_layout(showcase, contents)
+
+    theme_class_str = f" bg-{theme_color}" if theme_color else ""
+    class_str = f" {class_}" if class_ is not None else ""
```

Comment:
```suggestion
```

Reply:
Can not do this as `class_` has special treatment